src/utils/blockchainHelpers.js

```diff
@@ -344,6 +344,46 @@ async function getAllApplicationsInstances() {
   return Promise.all(whenCrowdsales).then(crowdsales => crowdsales.filter(crowdsale => crowdsale !== null))
 }
 
+async function getOwnerApplicationsInstancesForProxy() {
```

Comment:
May I suggest you a refactor for this method?

As suggested in this [SO Answer](https://stackoverflow.com/a/37576787), the way to go is with a map. I used a `reduce` to enforce the order, having the ability to list from newer to older the crowdsales created.

```javascript
async function getOwnerApplicationsInstancesForProxy() {
  const { web3 } = web3Store
  const proxiesRegistryContract = await attachToSpecificCrowdsaleContract('ProxiesRegistry')
  const accounts = await web3.eth.getAccounts()
  const execIDs = await proxiesRegistryContract.methods.getCrowdsalesForUser(accounts[0]).call()

  const { abi } = contractStore.MintedCappedProxy
  const mintedCapped = process.env[`${REACT_PREFIX}MINTED_CAPPED_APP_NAME`].toLowerCase()
  const dutchAuction = process.env[`${REACT_PREFIX}DUTCH_APP_NAME`].toLowerCase()

  return Promise.all(
    execIDs.reduce((promise, execID) => {
      return promise.then(async () => {
        try {
          const contractInstance = await attachToContract(abi, execID)
          const contractAppName = await contractInstance.methods.app_name.call()
          const appName = removeTrailingNUL(web3.utils.toAscii(contractAppName)).toLowerCase()

          if (appName.includes(mintedCapped) || appName.includes(dutchAuction)) {
            return { appName, execID }
          }
        } catch (error) {
          logger.error(error)
        }
      })
    }, Promise.resolve())
  )
```

Reply:
@fernandomg  Unfortunately, such refactor won't work because `return` doesn't return an iterable object: 
![screen shot 2018-07-13 at 13 52 47](https://user-images.githubusercontent.com/4341812/42688288-57f3cba6-86a4-11e8-9baa-41edc878f7aa.png)

I've implemented all useful and working parts from your suggestion. Thanks for sharing. Let's return to the final refactor of `getOwnerApplicationsInstancesForProxy` function in a separate PR.